### PR TITLE
If the session fails to be created get a new one

### PR DESCRIPTION
**Pull request by athompson-r7**
Some of the test that fetch a session and try to use it have been failing in a very random way.  The aim of this Pull Request is to make sure a session still exists after a period of time and if it doesn't try to get a new one.

### CODE REVIEW

File: lib/daemon_runner/session.rb [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+    class CreateSessionError < SessionError; end
+
     include Logger
 
     class << self
       attr_reader :session
 
       def start(name, **options)
         @session ||= Session.new(name, options).renew!
+        raise CreateSessionError, 'Failed to create session' if @sesison == false
```
Comment (dgreene-r7) [inline lib/daemon_runner/session.rb]: Typo! `@session != @sesison` :smile:

File: lib/daemon_runner/session.rb [hunk 2]
```diff
@@ -107,6 +114,18 @@ def destroy!
       Diplomat::Session.destroy(id)
     end
 
+    # Verify wheather the session exists after a period of time
+    def verify_session
+      logger.info(" - Wait until Consul session #{id} exists")
+      Timeout.timeout(2) do
+        sleep 1 if session_exist?
```
Comment (dgreene-r7) [inline lib/daemon_runner/session.rb]: If your goal is to determine whether the given session actually exists then:

1. 2 seconds is a pretty short period of time considering that calling `Diplomat::Session.list` wraps an HTTP request and could take an indeterminate amount of time.
2. expecting a timeout and rescuing it seems kind of, well, gross. (nb. Is this considered an anti-pattern in Ruby? I know in Python it's completely acceptable to use exceptions for flow control but I'll admit I'm not sure if Ruby is the same way.)

File: lib/daemon_runner/session.rb (hunk 1 above)
Comment (athompson-r7) [inline lib/daemon_runner/session.rb]: Typo fixed in 1d92cd0.

File: lib/daemon_runner/session.rb (hunk 2 above)
Comment (athompson-r7) [inline lib/daemon_runner/session.rb]: I have found the `/v1/session/list` api to be very flakey, sometimes session will stick sometimes they will not.  An immediate get to `/v1/session/list` will almost always return the session, but after some short period of time(less than the TTL) the session will be gone.

Yes I know this is gross, do you have suggestions?

> 2 seconds is a pretty short period of time

I didn't want to block the session create process for longer that necessary.

Comment (athompson-r7) [inline lib/daemon_runner/session.rb]: @dgreene-r7 @jmanero-r7 5f5ad5e might make a little more sense...

**Pull request merged**